Cauldron/Assets/Scripts/UI/Menu/SettingsMenu.cs

```diff
@@ -28,9 +70,92 @@ public void Close()
             gameObject.SetActive(false);
         }
 
+        private void LoadResolutionDropdown()
+        {
+            resolutions = Screen.resolutions;
+            resolutionDropdown.ClearOptions();
+            List<string> options = new List<string>();
+            int currentResolutionIndex = 0;
+            int setResolutionIndex = 0;
+
+            foreach (var res in resolutions)
+            {
+                options.Add(res.width + " x " + res.height);
+                if (res.width == Screen.width && res.height == Screen.height)
+                {
+                    setResolutionIndex = currentResolutionIndex;
+                }
+                currentResolutionIndex++;
+            }
+
+            resolutionDropdown.AddOptions(options);
+            resolutionDropdown.value = setResolutionIndex;
+            resolutionDropdown.RefreshShownValue();
+        }
+
+        public void ChangeResolution(int resIndex)
+        {
+            Resolution newResolution = resolutions[resIndex];
+            Screen.SetResolution(newResolution.width, newResolution.height, fullscreenMode);
+            PlayerPrefs.SetInt(PrefKeys.ResolutionSettings, resIndex);
+        }
+
         private void ChangeVolume(string vca, float value, float max = 1)
         {
             RuntimeManager.GetVCA($"vca:/{vca}").setVolume(Mathf.Lerp(0, max, value));
+            UpdateSliderLabel(vca, value);
+            PlayerPrefs.SetFloat(PrefKeys.MusicValueSettings, music.value);
+            PlayerPrefs.SetFloat(PrefKeys.SoundsValueSettings, sounds.value);
+        }
+
+        private void ChangeFullscreenMode(bool set)
+        {
+            fullscreenMode = set;
+            PlayerPrefs.SetInt(PrefKeys.FullscreenModeSettings, fullscreenMode ? 1 : 0);
+            Screen.fullScreen = fullscreenMode;
+        }
+
+        private void LoadFullscreenMode()
+        {
+            if (PlayerPrefs.HasKey(PrefKeys.FullscreenModeSettings))
+            {
+                fullscreenMode = PlayerPrefs.GetInt(PrefKeys.FullscreenModeSettings) == 1;
+                toggleFullscreen.isOn = fullscreenMode;
+            }
+        }
+
+        private void UpdateSliderLabel(string vca, float value)
+        {
+            string labelValue = Mathf.RoundToInt(Mathf.Lerp(0, 1, value) * 100) + "%";
```

Comment:
This line is kinda strange, because Lerp(0, 1, value) is actually equal to 'value'.